readthedocs/projects/forms.py

```diff
@@ -202,6 +213,26 @@ class Meta(object):
             # 'num_major', 'num_minor', 'num_point',
         )
 
+    def __init__(self, *args, **kwargs):
+        super(ProjectAdvancedForm, self).__init__(*args, **kwargs)
+
+        default_choice = (None, '-' * 9)
+        versions = self.instance.versions.values_list('slug', 'verbose_name')
+        self.fields['default_branch'].widget = forms.Select(
+            choices=[default_choice] + list(versions)
+        )
+
+        versions = (
+            self.instance.all_active_versions()
+            .filter(privacy_level=PUBLIC)
```

Comment:
This also doesn't hold up on our commercial hosting. It might be best to leave this sort of logic out of a query entirely, but at very least, it should only be checking this conditionally, if private privacy is not allowed.